### PR TITLE
A4A: Refactor expertise form.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR modifies the Partner directory expertise form to use the official data structure and the existing 'Services' and 'Products' selector component.

## Proposed Changes

* Update useExpertiseForm to use the `AgencyDirectoryApplication` type as the formData type. Update some functions to match the new data structure and remove unnecessary constants to represent fields.
* Update the Expertise form component to use the **'Services'** and **'Products'** selector component.
* Add initial CTA functionality in the Expertise form.
* Make the Expertise form aware of initial data.

## Testing Instructions

* Use the A4A live link and go to `/partner-directory/agency-expertise`
* Confirm that the form is working as expected.
   * Confirm the Services and Products field uses the new selector.
   * Confirm that the CTA button (disable state) works as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/partner-directory/types.ts [hunk 1]
```diff
@@ -1,13 +1,17 @@
+export type AgencyDirectoryApplicationStatus = 'pending' | 'in-progress' | 'completed';
+
+export type DirectoryApplicationType = 'wordpress' | 'jetpack' | 'woocommerce' | 'pressable';
+
 export interface AgencyDirectoryApplication {
-	status: 'pending' | 'in-progress' | 'completed';
+	status?: AgencyDirectoryApplicationStatus;
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/partner-directory/types.ts]: If I recall correctly, in the end, we leave this as a required parameter.

Comment (jkguidaven) [reply client/a8c-for-agencies/sections/partner-directory/types.ts]: Good catch. I've made the necessary adjustments to resolve this. 👍

File: client/a8c-for-agencies/sections/partner-directory/lib/agency-directory-application-form.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/*
+ * Check if the form has directories selected
+ */
+export function hasDirectory( formData: AgencyDirectoryApplication ): boolean {
+	return formData.directories.length > 0;
+}
+
+export function isValidApplicationForm( formData: AgencyDirectoryApplication ): boolean {
+	return hasServices( formData ) && hasProducts( formData ) && hasDirectory( formData );
+}
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/partner-directory/lib/agency-directory-application-form.tsx]: Really, do we need a library for this?

I'd add those helpers to the `agency-expertise/index.tsx` file, and if becomes bigger, I'd create a `helper` file in the same folder, `agency-expertise/`

Comment (cleacos) [reply client/a8c-for-agencies/sections/partner-directory/lib/agency-directory-application-form.tsx]: I'm going to take over this PR and add a commit

**Review by cleacos — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14398093

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @jkguidaven, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Submit my application`

Thank you in advance!

**Comment by jkguidaven** [comment]
@cleacos thanks for taking over on this PR 👍

File: client/a8c-for-agencies/sections/partner-directory/lib/agency-directory-application-form.tsx (hunk 2 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/partner-directory/lib/agency-directory-application-form.tsx]: @cleacos, good point. I wasn't sure really where to put this. But yeah `agency-expertise/index.tsx` looks good to me 👍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14398093).

**Pull request merged**